scripts/gdb/lvgl.py

```diff
@@ -0,0 +1,179 @@
+# Usage: source lvgl.py
+import argparse
+import gdb
+
+
+gdb.execute("set pagination off")
+gdb.write("set pagination off\n")
+gdb.execute("set python print-stack full")
+gdb.write("set python print-stack full\n")
+
+
+def lv_global():
+    return gdb.parse_and_eval("lv_global")
+
+
+def lv_display_get_default():
+    return lv_global()["disp_default"]
+
+
+def lv_screen_active():
+    disp = lv_display_get_default()
+    return disp["act_scr"] if disp else None
+
+
+def lv_obj_get_child_count(obj: gdb.Value):
+    return obj["spec_attr"]["child_cnt"] if obj["spec_attr"] else 0
+
+
+def lv_obj_get_child(obj: gdb.Value, index: int):
+    return obj["spec_attr"]["children"][index] if obj["spec_attr"] else None
+
+
+def lv_obj_get_class(obj: gdb.Value):
+    return obj["class_p"]
+
+
+def lv_class_get_name(clz: gdb.Value):
+    return clz["name"].string() if clz else ""
+
+
+def lv_obj_get_class_name(obj: gdb.Value):
+    class_p = lv_obj_get_class(obj)
+    return lv_class_get_name(class_p)
+
+
+def foreach_child(obj: gdb.Value):
+    for i in range(lv_obj_get_child_count(obj)):
+        yield lv_obj_get_child(obj, i)
+
+
+def foreach_obj_style(obj: gdb.Value):
+    LV_STYLE_PROP_INV = 0
+    LV_STYLE_PROP_ANY = 0xFF
+    count = obj["style_cnt"]
+    if count == 0:
+        return
+
+    styles = obj["styles"]
+    for i in range(count):
+        style = styles[i]["style"]
+        prop_cnt = style["prop_cnt"]
+        values_and_props = style["values_and_props"].cast(
+            gdb.lookup_type("lv_style_const_prop_t").pointer()
+        )
+        for j in range(prop_cnt):
+            prop = values_and_props[j]["prop"]
+            if prop == LV_STYLE_PROP_INV or prop == LV_STYLE_PROP_ANY:
+                continue
+            yield values_and_props[j]
+
+
+def dump_obj_info(obj: gdb.Value):
+    clzname = lv_obj_get_class_name(obj)
+    coords = obj["coords"]
+    coords = f"({int(coords['x1'])},{int(coords['y1'])},{int(coords['x2'])},{int(coords['y2'])})"
+
+    print(f"{clzname}@{hex(obj)} {coords}")
```

Comment:
What about printing `obj_id` too?

Reply:
Let's add it step by step. It's not an easy task since objid could be builtin id or platform dependent.